src/uproot/writing/_cascadetree.py

```diff
@@ -668,9 +676,10 @@ def extend(self, file, sink, data):
 
                 content = layout.content
                 offsets = numpy.asarray(layout.offsets)
+
                 if offsets[0] != 0:
                     content = content[offsets[0] :]
-                    offsets -= offsets[0]
+                    offsets = offsets - offsets[0]
```

Comment:
This is an important bug-fix. Don't let it get lost among the tweaks & tunes!

Reply:
Basically, it improves speed by not writing small baskets. So is there a way to set the min size per tree, such as 
```
uproot.newtree(..., flushsize=100000)
```
And then do:
```
file.extend(..., flush=False)
```
And this way, keep stuff in memory and only write to file when the basket is full. So is this (https://indico.cern.ch/event/840667/contributions/3527109/attachments/1908764/3153297/uproot-irisfellow-final.pdf) out of date for uproot4 now?